src/main/kotlin/org/rust/cargo/runconfig/command/CargoCommandType.kt

```diff
@@ -9,20 +9,18 @@ import com.intellij.openapi.project.Project
 import com.intellij.openapi.util.Key
 import org.rust.ide.icons.RustIcons
 
-class CargoCommandRunConfigurationType : ConfigurationTypeBase("CargoCommandRunConfiguration",
-                                                               "Cargo Command",
-                                                               "Cargo command run configuration",
-                                                               RustIcons.RUST) {
+class CargoCommandType : ConfigurationTypeBase("CargoCommandConfiguration",
+                                               "Cargo Command",
+                                               "Cargo command run configuration",
+                                               RustIcons.RUST) {
     init {
         addFactory(object : ConfigurationFactory(this) {
             override fun createTemplateConfiguration(project: Project): RunConfiguration =
-                CargoCommandConfiguration(project, "Cargo", this@CargoCommandRunConfigurationType)
+                CargoCommand(project, "Cargo", this@CargoCommandType)
 
             override fun configureBeforeRunTaskDefaults(providerID: Key<out BeforeRunTask<BeforeRunTask<*>>>,
                                                         task: BeforeRunTask<out BeforeRunTask<*>>) {
                 if (providerID == CompileStepBeforeRun.ID) {
-                    // We don't use jps, so we don't need to execute `Make` task
-                    // before run configuration is executed
                     task.isEnabled = false
                 }
```

Comment:
This `ConfigurationFactory` is repeated in all three different configurations. I think this duplication can be avoided by introducing `CargoConfigurationFactory` class. It will also allow us to preserve the comment ;)